### ISSUE TITLE
Oauth no longer works, how do I use cookies on a headless server?

### ISSUE THREAD

**Issue by twilson90** [yt-dlp/yt-dlp#11585]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I can't see anywhere in the KNOWN ISSUES/FAQ section how to set the cookie if I don't have access to a browser on a remote machine.
Also cookies expire, what can I do to prevent having to keep updating the cookie once it's assigned?
Thanks.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by october262** [comment]
try this -  When using yt-dlp on a remote machine without access to a browser, you can’t directly extract cookies from a browser. However, there are alternative methods to obtain cookies:

Manually obtain cookies:
If you have access to the same network or machine as the browser, you can manually extract cookies from the browser’s cookie file (e.g., cookies.txt or Network/Cookies folder).
Copy the cookies file to the remote machine and use yt-dlp with the --cookies option, specifying the path to the cookies file.
Use a third-party tool to extract cookies:
Tools like get-cookies.txt-locally (mentioned in the search results) can help you extract cookies from a browser’s cookie file without needing to access the browser directly.
You can use these tools to generate a cookies file, which you can then use with yt-dlp on the remote machine.
Implement a custom cookie extraction solution:
If none of the above options work for you, you can develop a custom script or tool to extract cookies from the browser’s cookie file or network traffic.
This approach requires programming expertise and may involve using libraries like requests or BeautifulSoup to parse HTTP traffic.

**Comment by october262** [comment]
and for cookies expiring, try - Keep YouTube Cookies Persistent
Cookies used by yt-dlp for YouTube and other platforms are typically session-based and designed to expire after a certain period. This is a security measure to prevent cookie theft and abuse. When a cookie expires, yt-dlp will re-fetch the necessary cookies for the next request.

Workarounds
Keep your browser session open: As mentioned in some Reddit threads (e.g., r/youtubedl), keeping your browser session open while using yt-dlp can help cache necessary cookies, reducing the need for frequent re-fetching.
Use a cookie manager: Tools like ChromeCookiesView (mentioned in r/youtubedl) can help extract and manage cookies from your browser. You can then use these cookies with yt-dlp.
Cache cookies in a database: Some users have suggested creating a cookies database (e.g., in ~/.config/yt-dlp) to store and reuse cached cookies. This approach is not implemented in yt-dlp by default, but you can explore custom scripting or plugins to achieve this.

**Comment by twilson90** [comment]
OK I've got it working by exporting a cookies.txt to the remote server and configuring yt-dlp.conf using a chrome profile that I will not use again (assuming it might invalidate the old cookie values).
I'm a bit confused by your second reply though.
I googled 'Keep YouTube Cookies Persistent' and got no hits.
How would yt-dlp 're-fetch the necessary cookies for the next request'?
Essentially I want to set this up once and never have to manually re-upload cookies or configure yt-dlp again (or until YouTube adds an extra hoop to jump through)
